### PR TITLE
use FSRS4.5+ forgetting curve for algos that do not have implicitly defined curve

### PR DESCRIPTION
**Pull request by Gilfaro**
Use FSRS4.5+ forgetting curve for algos that do not have implicitly defined curve.
This improves all of the NN based algos, but you will need to pretrain again.
Similar thing could be done for HLR which improves it as well, but it wouldn't be half life anymore, not sure if you would want to have 2 variants to see how forgetting curve alone impacts the same algo.

### CODE REVIEW

**Comment by Expertium** [comment]
Yeah, let's not touch HLR, since that's how it was originally defined. No problem with adding the new curve to GRU and Transformer, though.
EDIT: and don't change NN-17 either, it's meant to be similar to [SM-17](https://supermemo.guru/wiki/Algorithm_SM-17).

File: other.py [hunk 1]
```diff
@@ -875,7 +875,7 @@ def forward(self, src):
         return output, None
 
     def forgetting_curve(self, t, s):
```
Comment (L-M-Sherlock) [inline other.py:877]: This `forgetting_curve` is unused in the prediction of `GRU_P`.

Comment (Gilfaro) [reply other.py:877]: That is correct, it doesn't even have such function, only Transformer and RNN classes are modified

**Review by L-M-Sherlock — APPROVED** [review]
LGTM

**Comment by L-M-Sherlock** [comment]
I will re-run the benchmark soon.

**Pull request merged**